plugins/woocommerce/src/StoreApi/Schemas/V1/CheckoutSchema.php

```diff
@@ -243,7 +243,7 @@ protected function get_checkout_response( \WC_Order $order, PaymentResult $payme
 			'shipping_address'   => (object) $this->shipping_address_schema->get_item_response( $order ),
 			'payment_method'     => $order->get_payment_method(),
 			'payment_result'     => $payment_result,
-			'additional_fields'  => $this->get_additional_fields_response( $order ),
+			'additional_fields'  => (object) $this->get_additional_fields_response( $order ),
```

Comment:
when no fields are present, this would be an array, causing logic issues down the road.

Reply:
Common pattern to just cast at the schema level (last step) to avoid any short-circuit doing something else, but that function only has one returned, moved it there.